framework/CodeInterfaces/Saphire/SaphireData.py

```diff
@@ -0,0 +1,112 @@
+
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import numpy as np
+import os
+import copy
+from sklearn import neighbors
+import re
+"""
+Created on July 12, 2018
+
+@author: wangc
+"""
+
+class SaphireData:
+  """
+    Class that parses output of SAPHIRE outputs and write a RAVEN compatible CSV
+  """
+
+  def __init__(self, outFiles):
+    """
+      Initialize the class
+      @ In, outFiles, list, list of output files of SAPHIRE
+      @ Out, None
+    """
+    self.headerNames = []
+    self.outData = []
+    for outFile in outFiles:
+      outFileName, outFileType = outFile[0], outFile[1]
+      if outFileType == 'uncertainty':
+        headers, data = self.getUncertainty(outFileName)
+        self.headerNames.extend(headers)
+        self.outData.extend(data)
+      elif outFileType == 'importance':
+        headers, data = self.getImportance(outFileName)
+        self.headerNames.extend(headers)
+        self.outData.extend(data)
+      elif outFileType == 'quantiles':
+        print("File:",outFileName, "with type", outFileType, "is not implemented yet! Skipping" )
+        pass
+      else:
+        raise IOError('The output file', outFileName, 'with type', outFileType, 'is not supported yet!')
+
+  def getUncertainty(self, outName):
+    """
+      Method to extract the uncertainty information of Event Tree or Fault Tree from SAPHIRE output files
+      @ In, outName, string, the name of output file
+      @ Out, headerNames, list, list of output variable names
```

Comment:
You should indicate that the output is a tuple, and after you clarify what the tuple contains

Reply:
changed.